handlers/pipeline.go

```diff
@@ -183,12 +184,66 @@ func PipelineUpdate(c echo.Context) error {
 
 		// Update active pipelines
 		pipeline.GlobalActivePipelines.ReplaceByName(currentName, foundPipeline)
+	}
+
+	// Check if the periodic scheduling has been changed.
+	if !stringSliceEqual(foundPipeline.PeriodicSchedules, p.PeriodicSchedules) {
+		// We prevent side effects here and make sure
+		// that no scheduling is already running.
+		if foundPipeline.CronInst != nil {
+			foundPipeline.CronInst.Stop()
+		}
+		foundPipeline.CronInst = cron.New()
+
+		// Iterate over all cron schedules.
+		for _, cron := range p.PeriodicSchedules {
+			err := foundPipeline.CronInst.AddFunc(cron, func() {
+				_, err := schedulerService.SchedulePipeline(&foundPipeline, []gaia.Argument{})
+				if err != nil {
+					gaia.Cfg.Logger.Error("cannot schedule pipeline from periodic schedule", "error", err, "pipeline", foundPipeline)
+					return
+				}
+
+				// Log scheduling information
+				gaia.Cfg.Logger.Info("pipeline has been automatically scheduled by periodic scheduling:", "name", foundPipeline.Name)
+			})
+
+			if err != nil {
+				return c.String(http.StatusBadRequest, err.Error())
+			}
+		}
 
+		// Update pipeline in store
+		err := storeService.PipelinePut(&foundPipeline)
+		if err != nil {
+			return c.String(http.StatusInternalServerError, err.Error())
+		}
+
+		// Start schedule process.
+		foundPipeline.CronInst.Start()
+
+		// Update active pipelines
+		pipeline.GlobalActivePipelines.Replace(foundPipeline)
 	}
 
 	return c.String(http.StatusOK, "Pipeline has been updated")
 }
 
+// stringSliceEqual is a small helper function
+// which determines if two string slices are equal.
+func stringSliceEqual(a, b []string) bool {
```

Comment:
Why not use reflect.DeepEqual here?

Reply:
Good point. I've learned that `reflect.DeepEqual` is super slow and you should only use that for tests (e.g. https://programming.guide/go/compare-slices.html). Would love to hear your opinion 😉

Reply:
It depends on one often do you think this function is being called versus the complexity that it adds in terms of things that you need to cover by tests. :) If you are happy with it, it's okay. :)